### PR TITLE
Update SQL timestamp functions and casting operations

### PR DESCRIPTION
**Pull request by sanderson**
Closes https://github.com/influxdata/docs-v2/issues/5331

- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud-dedicated/query-data/sql/cast-types.md [hunk 1]
```diff
@@ -303,7 +303,49 @@ SQL supports casting the following to a boolean:
...
-- **Integers**: Parses the integer as a Unix nanosecond timestamp and returns
+- **Integers**: Parses the integer as a Unix _second_ timestamp and returns
   the equivalent timestamp.
 - **Unsigned integers**: Parses the unsigned integer as a Unix nanosecond timestamp
   and returns the equivalent timestamp.
+
+{{% note %}}
+#### Cast Unix nanosecond timestamps to a timestamp type
+
+To cast a Unixnanosecond timestamp to a timestamp type, first cast the numeric
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/query-data/sql/cast-types.md:314]:
```suggestion
To cast a Unix nanosecond timestamp to a timestamp type, first cast the numeric
```

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 2]
```diff
@@ -120,7 +121,9 @@ Calculates time intervals and returns the start of the interval nearest to the s
 Use `date_bin` to downsample time series data by grouping rows into time-based "bins" or "windows"
 and applying an aggregate or selector function to each window.
 
-For example, if you "bin" or "window" data into 15 minute intervals, an input timestamp of `2023-01-01T18:18:18Z` will be updated to the start time of the 15 minute bin it is in: `2023-01-01T18:15:00Z`.
+For example, if you "bin" or "window" data into 15 minute intervals, an input
+timestamp of `2023-01-01T18:18:18Z` will be updated to the start time of the
+15 minute bin it is in: `2023-01-01T18:15:00Z`.
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:126]:
```suggestion
For example, given an input
timestamp of `2023-01-01T18:18:18Z`, if you bin data into 15-minute intervals, the output timestamp is `2023-01-01T18:15:00Z`--the start of the 15-minute bin that includes the input.
```

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 3]
```diff
@@ -693,24 +691,49 @@ to_timestamp_micros(expression)
...
+| :------------------------------------------- |
+| 2024-01-01T00:00:00.000001Z                  |
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_timestamp_nanos
+
+Converts a value to RFC3339 nanosecond timestamp format (`YYYY-MM-DDT00:00:00.000000000Z`).
+Supports timestamp, integer, and unsigned integer types as input.
+Integers and unsigned integers are parsed as [Unix nanoosecond timestamps](/influxdb/cloud-dedicated/reference/glossary/#unix-timestamp)
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:707]:
```suggestion
Integers and unsigned integers are parsed as [Unix nanosecond timestamps](/influxdb/cloud-dedicated/reference/glossary/#unix-timestamp)
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 4]
```diff
@@ -697,24 +692,49 @@ to_timestamp_micros(expression)
...
+| :------------------------------------------- |
+| 2024-01-01T00:00:00.000001Z                  |
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_timestamp_nanos
+
+Converts a value to RFC3339 nanosecond timestamp format (`YYYY-MM-DDT00:00:00.000000000Z`).
+Supports timestamp, integer, and unsigned integer types as input.
+Integers and unsigned integers are parsed as [Unix nanoosecond timestamps](/influxdb/cloud-serverless/reference/glossary/#unix-timestamp)
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:708]:
```suggestion
Integers and unsigned integers are parsed as [Unix nanosecond timestamps](/influxdb/cloud-serverless/reference/glossary/#unix-timestamp)
```

File: content/influxdb/clustered/query-data/sql/cast-types.md [hunk 5]
```diff
@@ -303,7 +303,49 @@ SQL supports casting the following to a boolean:
...
-- **Integers**: Parses the integer as a Unix nanosecond timestamp and returns
+- **Integers**: Parses the integer as a Unix _second_ timestamp and returns
   the equivalent timestamp.
 - **Unsigned integers**: Parses the unsigned integer as a Unix nanosecond timestamp
   and returns the equivalent timestamp.
+
+{{% note %}}
+#### Cast Unix nanosecond timestamps to a timestamp type
+
+To cast a Unixnanosecond timestamp to a timestamp type, first cast the numeric
```
Comment (jstirnaman) [inline content/influxdb/clustered/query-data/sql/cast-types.md:314]:
```suggestion
To cast a Unix nanosecond timestamp to a timestamp type, first cast the numeric
```

**Review by jstirnaman — APPROVED** [review]
Just some typo fixes and a suggestion.

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 6]
```diff
@@ -693,24 +688,49 @@ to_timestamp_micros(expression)
...
+| :------------------------------------------- |
+| 2024-01-01T00:00:00.000001Z                  |
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## to_timestamp_nanos
+
+Converts a value to RFC3339 nanosecond timestamp format (`YYYY-MM-DDT00:00:00.000000000Z`).
+Supports timestamp, integer, and unsigned integer types as input.
+Integers and unsigned integers are parsed as [Unix nanoosecond timestamps](/influxdb/clustered/reference/glossary/#unix-timestamp)
```
Comment (sanderson) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:704]:
```suggestion
Integers and unsigned integers are parsed as [Unix nanosecond timestamps](/influxdb/clustered/reference/glossary/#unix-timestamp)
```

**Pull request merged**